doc/man3/SSL_set1_cache_id.pod

```diff
@@ -0,0 +1,105 @@
+=pod
+
+=head1 NAME
+
+SSL_set1_cache_id,
+SSL_get0_cache_id,
+SSL_SESSION_set1_cache_id,
+SSL_SESSION_get0_cache_id,
+SSL_get1_previous_client_session - use application-defined cache identifier for session caching
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ int SSL_set1_cache_id(SSL *s, const unsigned char *data, size_t len);
+ int SSL_get0_cache_id(const SSL *s, const unsigned char **data, size_t *len);
+
+ int SSL_SESSION_set1_cache_id(SSL_SESSION *ss, const unsigned char *data, size_t len);
+ int SSL_SESSION_get0_cache_id(const SSL_SESSION *ss, const unsigned char **data, size_t *len);
+
+ int SSL_get1_previous_client_session(SSL *s, int flags);
+
+=head1 DESCRIPTION
+
+SSL_set1_cache_id() sets the cache identifer specified by B<data> and B<len>
+into B<s>. This cache identifier is application-defined, and can be a domain
+name, IP address or other identifier.
+
```

Comment:
There seems like there could be an opportunity to streamline the API a little here. There seems to be some overlap with `SSL_set_tlsext_host_name`. I guess frequently the `cache_id` will just be the hostname, possibly with a port as well. Is it worth having a function that does both of these things?

Reply:
The "cache id" is a value that can include anything the user chooses, it can be "IPADDR:PORT", "example.com:port", etc. The port can be made part of the cache ID, to keep it as generic as possible, and not to require a bunch of different parameters that users may not care about.

Reply:
So, while those functions are similar, the cache_id is meant to be more generic (e.g. a structure, or  other structured information), and will likely include more information than what is put into `SSL_set_tlsext_host_name`.